version: "3.8"
services:
  django:
    build: .
    container_name: django
    command: python manage.py runserver 0.0.0.0:8000
    volumes:
      - ./:/usr/src/app/
    ports:
      - "8000:8000"
    environment:
      - DEBUG=1
      - DJANGO_ALLOWED_HOSTS=localhost
      - CELERY_BROKER=redis://redis:6379/0
      - CELERY_BACKEND=redis://redis:6379/0
    depends_on:
      - pgdb
      - redis
  celery:
    build: .
    command: celery -A main worker -B -l INFO
    volumes:
      - ./:/usr/src/app/
    depends_on:
      - django
      - redis
  pgdb:
    image: "postgres:latest"
    container_name: pgdb
    environment:
      - POSTGRES_DB=KZshops
      - POSTGRES_USER=postgres
      - POSTGRES_PASSWORD=123456789i
      - POSTGRES_HOST=localhost
    ports:
      - "5444:5432"
    volumes:
      - postgres_data:/var/lib/postgresql/data/
  redis:
    image: "redis:alpine"
    container_name: redis
    ports:
      - '6380:6379'
volumes:
  postgres_data:




